src/libs/OptionsListUtils.js

```diff
@@ -636,7 +637,57 @@ function hasEnabledOptions(options) {
 }
 
 /**
- * Build the options for the category tree hierarchy via indents
+ * Sorts categories using a simple object.
+ * It builds an hierarchy (based on an object), where each category has a name and other keys as subcategories.
+ * Via the hierarchy we avoid duplicating and sort categories one by one. Subcategories are being sorted alphabetically.
+ *
+ * @param {Object<String, {name: String, enabled: Boolean}>} categories
+ * @returns {Object[]}
+ */
+function sortCategories(categories) {
+    const hierarchy = {};
+
+    _.each(categories, (category) => {
+        const path = category.name.replaceAll(CONST.PARENT_CHILD_SEPARATOR, '.');
+        const existedValue = lodashGet(hierarchy, path, {});
+
+        lodashSet(hierarchy, path, {
+            ...existedValue,
+            name: category.name,
+        });
+    });
+
+    const flatHierarchy = (initialHierarchy) =>
```

Comment:
Similarly, it'd be nice to add a comment explaining what this is doing.

Reply:
Done - 9fa9bca.

Reply:
Thank you!